### ISSUE TITLE
--outputPath always put outputs in relative path even when the given path is an absolute path.

### ISSUE THREAD

**Issue by pillsilly** [nrwl/nx#18262]
### Current Behavior

` npx nx build mylib --outputPath=/my/aboslute/path`

this will not result nx to generate under /my/aboslute/path at all

instead a folder called "/my/aboslute/path" relative to the command execution path would be used.

### Expected Behavior

nx should respect what's given to the outputPath and generate files under the path even it's an absolute path.

### Steps to Reproduce

1.npx nx build `mylib` --outputPath=`/my/aboslute/path`

replace quoted parts to any of your nx project's

### Nx Report

```shell
No Nx Report
```

### Failure Logs

```shell
No
```

### Operating System

- [ ] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by AgentEnder** [comment]
Can you provide the target configuration for mylib:build and `nx report` please?

**Comment by pillsilly** [comment]
> 

those are kept default since workspace was initiated via nx cli. 
so I think they are irrelevant to the RC.

Also ,
reading another doc -- https://nx.dev/extending-nx/recipes/generator-options
`        "description": "The output path of the application's generated HTML index file. The full provided path will be used and will be considered relative to the application's configured output path."
`
got me suspect that NX did it intentionally,

so it could be yet another _it's not a bug, it's A feature_.

if that's true, I wonder if it really contains any rationality about `The full provided path will be used and will be considered relative` 


allow me explain a bit about  my use-case of nx build :
1. I have a project A, it rely on a lib X, produced by project B which is Nx based.
2. I develop them both locally and I sometimes want to test X's new build locally too.

I tried 3 ways of doing that
1. copy X's build which is by default under B's directory, to the corresponding node_module directory under A.
2. npm link X's build as A's node module -- not worked, as npm link somehow doesn't recognize correctly the relative path like that 
3. run `nx build` under X with specifying the destination to be the A's corresponding node_module package directory -- not working as I described in this thread.


by the way #3 since the best solution because it's more intuitive, if Nx could resolve the absolute path just as an absolute path.
